sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3281,6 +3282,25 @@ def teardown_no_lock(self,
         config = common_utils.read_yaml(handle.cluster_yaml)
         cluster_name = handle.cluster_name
         use_tpu_vm = config['provider'].get('_has_tpus', False)
+
+        # Use the new provisioner for AWS.
+        if isinstance(cloud, clouds.AWS):
+            # TODO(suquark): An alternative approach is to move cloud
+            #  termination logic into cloud classes (for example,
+            #  "clouds.AWS" here). Currently I just use the code from
+            #  the new provisioner, which I believe benefits further
+            #  development of "SkyPilot as a service". Let's see how
+            #  this goes.
+            region = config['provider']['region']
+            if not self.run_on_head(handle, 'ray stop --force'):
+                logger.warning(
+                    'failed to take down Ray autoscaler on the head node')
```

Comment:
```suggestion
                    'Failed to take down Ray autoscaler on the head node. '
                    'It might be because the cluster's head node has already '
                    'been terminated. It is fine to skip this.')
```